### PR TITLE
189 pintify groove to profile

### LINKED ISSUE #189: pintify groove to_profile methods
Use pint syntax in all `IsoGroove.to_profile` implementations

reference: https://github.com/BAMWelDX/weldx/pull/142/files#diff-2910750b7788fa22f0ced6d6a9dfb3a2c50911000150af082e568b78d99866b6

### PR DESCRIPTION
**Pull request by mark141**
## Changes

Pint syntax is now used in all `IsoGroove.to_profile` implementations.

reference:
https://github.com/BAMWelDX/weldx/pull/142/files#diff-2910750b7788fa22f0ced6d6a9dfb3a2c50911000150af082e568b78d99866b6

## Related Issues

Closes #189 

## Checks

- [x] updated CHANGELOG.md
- [x] updated tests
- [ ] updated doc/
- [ ] update example/tutorial notebooks

### CODE REVIEW

File: weldx/welding/groove/iso_9692_1.py [hunk 1]
```diff
@@ -240,19 +240,22 @@ def to_profile(self, width_default: pint.Quantity = Q_(5, "mm")) -> geo.Profile:
              pint.Quantity (Default value = Q_(5, "mm"))
 
         """
-        t = self.t.to(_DEFAULT_LEN_UNIT).magnitude
-        b = self.b.to(_DEFAULT_LEN_UNIT).magnitude
-        width = width_default.to(_DEFAULT_LEN_UNIT).magnitude
+        t = self.t
+        b = self.b
+        width = width_default
```
Comment (CagtayFabry) [inline weldx/welding/groove/iso_9692_1.py:245]: I think we can shorten this to one line
```suggestion
        t, b, width  = self.t, self.b, width_default
```

Same for the other assignments

File: weldx/welding/groove/iso_9692_1.py [hunk 2]
```diff
@@ -240,19 +240,22 @@ def to_profile(self, width_default: pint.Quantity = Q_(5, "mm")) -> geo.Profile:
...
 
         # x-values
-        x_value = [-width, 0, 0, -width]
+        x_value = np.stack((-width, 0, 0, -width))
+
         # y-values
-        y_value = [0, 0, t, t]
+        y_value = Q_([0, 0], "mm")
+        y_value = np.append(y_value, t)
+        y_value = np.append(y_value, t)
```
Comment (CagtayFabry) [inline weldx/welding/groove/iso_9692_1.py:253]: this should work with `np.stack`
```suggestion
        y_value = np.stack((0,0,t,t))
```

please check the rest of the code for similar cases

File: weldx/welding/groove/iso_9692_1.py [hunk 3]
```diff
@@ -513,13 +516,13 @@ def to_profile(self, width_default: pint.Quantity = Q_(2, "mm")) -> geo.Profile:
              pint.Quantity (Default value = Q_(2, "mm"))
 
         """
-        t = self.t.to(_DEFAULT_LEN_UNIT).magnitude
+        t = self.t
         alpha = self.alpha.to("rad").magnitude
         beta = self.beta.to("rad").magnitude
```
Comment (CagtayFabry) [inline weldx/welding/groove/iso_9692_1.py:521]: I think `pint` should work fine with `np.cos` etc. ? Have you tried @mark141

File: weldx/welding/groove/iso_9692_1.py [hunk 4]
```diff
@@ -1575,19 +1585,19 @@ def _helperfunction(segment, array) -> geo.Shape:
...
                 # begin
-                array[1][counter],
+                array[1][counter].to(_DEFAULT_LEN_UNIT).magnitude,
                 # end
-                array[1][counter + 2],
+                array[1][counter + 2].to(_DEFAULT_LEN_UNIT).magnitude,
                 # circle center
-                array[1][counter + 1],
+                array[1][counter + 1].to(_DEFAULT_LEN_UNIT).magnitude,
             ]
```
Comment (CagtayFabry) [inline weldx/welding/groove/iso_9692_1.py:1601]: this should just be
```python
arr0 = array[0][[counter, counter + 2, counter +1]]
arr1 = array[1][[counter, counter + 2, counter +1]]
arr = np.vstack((arr0,arr1))
```
convert to magnitude after if necessary

when working with `pint` try to avoid list wherever possible, just stick to `numpy` style indexing and array operations

**Review by CagtayFabry — CHANGES_REQUESTED** [review]
good job, code can still be simplified in a few places 🤖

**Comment by mark141** [comment]
This should fix your requested changes and i added the same functionallities to the Froontal Face Grooves.

The "bug" of adding a line Segment/arc Segment is no longer needed since it is resolved with using the correct numpy syntax.

**Review by CagtayFabry — APPROVED** [review]
👍
